### DISCUSSION TITLE
Figure width and height attributes per output format

### DISCUSSION THREAD

**Discussion by rgaiacs** [quarto-dev/quarto-cli#3578] [Q&A] [answered]
When creating a web and print version of the document, I usually want to define different width and height attributes per output format. This is more relevant when I'm working with large figures.

Consider the following minimal working example,

``````
---
title: "Minimal Working Example"
format:
  docx: default
  html: default
---

Foo

![Awesome Logo](R_logo.svg.png){#fig-1 width="6.5cm"}

Bar
``````

It produces the HTML

![Screenshot from 2022-12-06 13-05-38](https://user-images.githubusercontent.com/1506457/205821482-8c002420-d5e9-4516-b3cb-4c68c2c74b92.png)

and the DOCX

![Screenshot from 2022-12-06 13-04-17](https://user-images.githubusercontent.com/1506457/205821316-c908cbcd-f547-41fd-92a6-eb14064105ed.png)

where the image has width of 6.5cm. How can I set up the width only for the DOCX document?

Will be great if Quarto supports something like

```
![Awesome Logo](R_logo.svg.png){#fig-1 doc-width="6.5cm"}
```

were `doc-` attributes are used only when building DOCX documents.

**Comment by cscheid** [comment, marked as answer]
Take a look at [conditional content](https://quarto.org/docs/authoring/conditional.html).

**Reply by rgaiacs** [reply]
I'm aware of conditional content but what I'm looking is conditional style. Think of [CSS's `@media`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media).

**Reply by cscheid** [reply]
I understand. But that's not how you do this in pure quarto markdown. The way to do that in pure quarto is to use conditional content, one entry for docx (where you specify the width), and another for html (where you don't).

If you don't want to do it in your markdown, you can set up a filter that only runs in docx and use that to affect behavior.
